modules/nixos/boot.nix

```
@@ -0,0 +1,10 @@
+{
+  config,
+  lib,
+  ...
+}:
+{
+  boot.loader.grub.efiSupport = lib.mkIf (config.facter.report.uefi.supported or null != null) (
```

Since it's possible to also enable uefi support in grub on bios system (to make it more portable), I would rather do:

```suggestion
  boot.loader.grub.efiSupport = lib.mkIf (config.facter.report.uefi.supported or false) (lib.mkDefault true);
```

---

Please also add an "detected" option like what we do for the rest, because this is what a layer we introduce for debugging and to distinguish from what facter detects vs how it affects the system.